pkg/runtime/runtime.go

```diff
@@ -2955,17 +2951,51 @@ func (a *DaprRuntime) WaitUntilShutdown() error {
 	return <-a.shutdownC
 }
 
-// Returns the component updated with the secrets applied.
-// If the component references a secret store that hasn't been loaded yet, it returns the name of the secret store component as second returned value.
-func (a *DaprRuntime) processComponentSecrets(component componentsV1alpha1.Component) (componentsV1alpha1.Component, string) {
+// Interface that applies to both Component and HTTPEndpoint resources.
+type resourceWithMetadata interface {
+	Kind() string
+	GetName() string
+	GetNamespace() string
+	GetSecretStore() string
+	NameValuePairs() []sharedapi.NameValuePair
+}
+
+func isEnvVarAllowed(key string) bool {
+	key = strings.ToUpper(key)
+	switch {
+	case key == "APP_API_TOKEN":
+		return false
+	case strings.HasPrefix(key, "DAPR_"):
+		return false
+	default:
+		return true
+	}
+}
```

Comment:
This is the same allowlist [implemented in the local.env secret store](https://github.com/dapr/components-contrib/pull/2511). It denies access to env vars in the sidecar container that could be dangerous to expose, such as `DAPR_API_TOKEN`, `DAPR_CERT_KEY`, etc

Reply:
This is an denylist, not allowlist. The sidecar injector can pass the list of keys from `env` as `DAPR_ALLOWED_ENV_KEYS` that this code can parse and use as an allowlist if it is set.

Reply:
I can add an allowlist but this denylist is a requirement for us for security reasons.

Reply:
Should have clarified that the denylist is needed for standalone mode where we can't have an injector adding an allowlist.